retrieval/relabel.go

```diff
@@ -62,6 +62,19 @@ func relabel(labels clientmodel.LabelSet, cfg *config.RelabelConfig) (clientmode
 		hasher.Write([]byte(val))
 		mod := hasher.Sum64() % cfg.Modulus
 		labels[cfg.TargetLabel] = clientmodel.LabelValue(fmt.Sprintf("%d", mod))
+	case config.RelabelMapLabels:
+		out := clientmodel.LabelSet{}
```

Comment:
This may also be used in `metric_relabel_configs` which are expensive as they are. Let's allocate the map in the correct size immediately.